src/Blue/Tools/Api/Client.php

```diff
@@ -225,24 +218,33 @@ public function setLogger(LoggerInterface $logger)
     }
 
     /**
-     * Returns the specified request option or all options if none specified
+     * Returns the specified request option or all options if none specified.
+     *
      * @param null $keyOrPath
+     *
      * @return array|mixed|null
      */
     public function getRequestOption($keyOrPath = null)
     {
-        return $this->guzzleClient->getDefaultOption($keyOrPath);
+        return $this->guzzleClient->getConfig($keyOrPath);
     }
 
     /**
-     * Sets a request option for future requests
-     * @param $keyOrPath
+     * Sets a request option for future requests.
+     *
+     * @param $key
      * @param $value
+     *
      * @return $this
      */
-    public function setRequestOption($keyOrPath, $value)
+    public function setRequestOption($key, $value)
     {
-        $this->guzzleClient->setDefaultOption($keyOrPath, $value);
+        $config = array_merge(
+            $this->guzzleClient->getConfig(),
+            [$key => $value]
+        );
+
+        $this->guzzleClient = new GuzzleClient($config);
         return $this;
```

Comment:
Guzzle no longer allows you to change config options after you've created the client, so this is the best I can do if I want to maintain backwards compatibility. The only real issue here is this:

```
$firstGuzzleClient = $client->getGuzzleClient();
$client->setRequestOption('abc', 'defg');
$secondGuzzleClient = $client->getGuzzleClient();
// $firstGuzzleClient != $secondGuzzleClient
```